### PR TITLE
[quant][eagermode][qat][test] Add numerical test for qat convert

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47376 [quant][eagermode][qat][test] Add numerical test for qat convert**

Summary:
For leaky_relu

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24734754](https://our.internmc.facebook.com/intern/diff/D24734754)

### CODE REVIEW

File: torch/quantization/fake_quantize.py [hunk 1]
```diff
@@ -212,8 +212,8 @@ def __init__(self,
         assert quant_min <= quant_max, 'quant_min should be less than or equal to quant_max'
         self.quant_min = quant_min
         self.quant_max = quant_max
-        self.register_buffer('scale', torch.tensor([scale]))
-        self.register_buffer('zero_point', torch.tensor([zero_point]))
+        self.register_buffer('scale', torch.tensor([scale], dtype=torch.float))
+        self.register_buffer('zero_point', torch.tensor([zero_point], dtype=torch.float))
```
Comment (vkuzo) [inline torch/quantization/fake_quantize.py:216]: qq, what's the context for this one?

Comment (jerryzh168) [reply torch/quantization/fake_quantize.py:216]: added when debugging, just to make it more explicit, is this needed?

Comment (vkuzo) [reply torch/quantization/fake_quantize.py]: makes sense, I was just wondering, lg

**Review by vkuzo — APPROVED** [review]

File: torch/quantization/fake_quantize.py (hunk 1 above)
Comment (jerryzh168) [reply torch/quantization/fake_quantize.py]: update: removed for now, looks like we didn't do that in fake_quantize.py right now, maybe we can add this back when we have better numerical test coverage on the qat conversion later.

**Pull request closed without merging**